examples/step-92/step-92.cc

```diff
@@ -0,0 +1,1091 @@
+    cells_solution_filename = "";
+    add_parameter("cells solution filename",
+                  cells_solution_filename,
+                  "filename for exporting cells solution");
+
+    add_parameter(
+      "materials",
+      materials_list,
+      "list of material definitions: \"name : ID : eps_r : kappa\"");
+    add_parameter("potentials",
+                  boundary_values_list,
+                  "list of potential definitions: \"name : ID : epot [V]\"");
+  }
+
+  template <int dim>
+  void ProblemParameters<dim>::finalize()
+  {
+    N_physicals = materials_list.size() + 1;
+
+    p_physicals = new material_definition[N_physicals];
+
+    int ix              = 0;
+    default_physical_ix = -1;
+    for (const auto &item : materials_list)
+      {
+        std::string  name  = std::get<0>(item);
+        unsigned int id    = std::get<1>(item);
+        double       eps_r = std::get<2>(item);
+        double       kappa = std::get<3>(item);
+
+        p_physicals[ix].name = name;
+
+        p_physicals[ix].material_id = id;
+        if (id == 0)
+          default_physical_ix = ix;
+
+        p_physicals[ix].epsilon = eps_r * epsilon_vacuum;
+        p_physicals[ix].kappa   = kappa;
+
+        ix++;
+      }
+
+    N_physicals = ix;
+
+    if (default_physical_ix < 0)
+      {
+        N_physicals++;
+        p_physicals[ix].name = "default";
+
+        default_physical_ix         = ix;
+        p_physicals[ix].material_id = 0;
+
+        p_physicals[ix].epsilon = epsilon_vacuum;
+        p_physicals[ix].kappa   = 0.0;
+      }
+
+    N_boundary_values = boundary_values_list.size();
+    p_boundary_values = new boundary_potential[N_boundary_values];
+
+    unsigned int        ipot = 0;
+    boundary_potential *t_p_opdata;
+    for (const auto &item : boundary_values_list)
+      {
+        std::string name      = std::get<0>(item);
+        auto        id        = std::get<1>(item);
+        double      potential = std::get<2>(item);
+
+        t_p_opdata              = &p_boundary_values[ipot];
+        t_p_opdata->material_id = id;
+        t_p_opdata->name        = name;
+        t_p_opdata->potential   = potential;
+
+        ipot++;
+      }
+  }
+
+  template <int dim>
+  ProblemParameters<dim>::~ProblemParameters()
+  {}
+
+  template <int dim>
+  std::string ProblemParameters<dim>::get_mesh_filename()
+  {
+    return mesh_filename;
+  }
+
+  template <int dim>
+  std::string ProblemParameters<dim>::get_cells_solution_filename()
+  {
+    return cells_solution_filename;
+  }
+
+  template <int dim>
+  typename ProblemParameters<dim>::material_definition
+  ProblemParameters<dim>::get_physical(types::material_id material)
+  {
+    for (unsigned int ix = 0; ix < N_physicals; ix++)
+      {
+        if (material == (p_physicals[ix].material_id))
+          {
+            return p_physicals[ix];
+          }
+      }
+
+    return p_physicals[default_physical_ix];
```

Comment:
I don't think that you'd expect/want to hit this line, so maybe add an assertion beforehand that triggers a failure if you reach it? Similar for the final return in `ProblemParameters::get_boundary_value()` below.